File: app/src/main/java/com/riders/thelab/ui/vectordrawables/VectorDrawablesActivity.java
package com.riders.thelab.ui.vectordrawables;

import android.annotation.SuppressLint;
import android.graphics.drawable.AnimatedVectorDrawable;
import android.graphics.drawable.Drawable;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.imageview.ShapeableImageView;
import com.riders.thelab.R;
import com.riders.thelab.data.local.bean.SnackBarType;

import butterknife.BindView;
import butterknife.ButterKnife;

@SuppressLint("NonConstantResourceId")
public class VectorDrawablesActivity extends AppCompatActivity {

    @BindView(R.id.iv_avd_google_io16)
    ShapeableImageView ivGoogleIO16;

    @BindView(R.id.iv_avd_progress_bar)
    ShapeableImageView ivProgressBar;

    AnimatedVectorDrawable googleIOAnimatedVectorDrawable;
    AnimatedVectorDrawable progressBarAnimatedVectorDrawable;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (!LabCompatibilityManager.isLollipop()) {
            UIManager.showActionInSnackBar(
                    this,
                    findViewById(android.R.id.content),
                    "Device running API blelow API 21 (Lollipop), canoont launch this activity",
                    SnackBarType.ALERT,
                    "LEAVE",
                    v -> finish());
        } else {
            setContentView(R.layout.activity_vector_drawables);

            ButterKnife.bind(this);
            getSupportActionBar().setTitle(getString(R.string.activity_title_vector_drawables));
        }
    }


    @SuppressLint("NewApi")
    @Override
    protected void onResume() {
        super.onResume();

        if (!LabCompatibilityManager.isLollipop()) {
            UIManager.showActionInSnackBar(
                    this,
                    findViewById(android.R.id.content),
                    "Device running API blelow API 21 (Lollipop), canoont launch this activity",
                    SnackBarType.ALERT,
                    "LEAVE",
                    v -> finish());
            /*new TheLabToast.Builder(this)
                    .setType(ToastTypeEnum.ERROR)
                    .setText("Your device's API cannot run this animation. Requires target API 21 (Lollipop)")
                    .show();*/
            return;
        }

        Drawable drawableGoogleIO16 = ivGoogleIO16.getDrawable();
        Drawable drawableProgressBar = ivProgressBar.getDrawable();

        if (drawableGoogleIO16 instanceof AnimatedVectorDrawable) {
            // AVD GoogleIO 16'
            googleIOAnimatedVectorDrawable = (AnimatedVectorDrawable) drawableGoogleIO16;
            googleIOAnimatedVectorDrawable.start();
        }

        if (drawableProgressBar instanceof AnimatedVectorDrawable) {
            // AVD Progress Bar
            progressBarAnimatedVectorDrawable = (AnimatedVectorDrawable) drawableProgressBar;
            progressBarAnimatedVectorDrawable.start();
        }
    }
}

File: app/src/main/java/com/riders/thelab/ui/vectordrawables/VectorDrawablesActivity.kt
package com.riders.thelab.ui.vectordrawables

import android.graphics.drawable.AnimatedVectorDrawable
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.MenuItem
import com.riders.thelab.core.ui.compose.base.BaseAppCompatActivity
import com.riders.thelab.databinding.ActivityVectorDrawablesBinding

class VectorDrawablesActivity : BaseAppCompatActivity() {

    lateinit var viewBinding: ActivityVectorDrawablesBinding

    private var googleIOAnimatedVectorDrawable: AnimatedVectorDrawable? = null
    private var progressBarAnimatedVectorDrawable: AnimatedVectorDrawable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBinding = ActivityVectorDrawablesBinding.inflate(layoutInflater)
        setContentView(viewBinding.root)
    }

    override fun onResume() {
        super.onResume()

        val drawableGoogleIO16: Drawable = viewBinding.ivAvdGoogleIo16.drawable
        val drawableProgressBar: Drawable = viewBinding.ivAvdProgressBar.drawable

        if (drawableGoogleIO16 is AnimatedVectorDrawable) {
            // AVD GoogleIO 16'
            googleIOAnimatedVectorDrawable = drawableGoogleIO16
            googleIOAnimatedVectorDrawable!!.start()
        }
        if (drawableProgressBar is AnimatedVectorDrawable) {
            // AVD Progress Bar
            progressBarAnimatedVectorDrawable = drawableProgressBar
            progressBarAnimatedVectorDrawable!!.start()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                backPressed()
            }
        }
        return true
    }

    override fun backPressed() {
        finish()
    }
}
